### DISCUSSION TITLE
Deprecated feature removal plans

### DISCUSSION THREAD

**Discussion by MarshallAsch** [liquidctl/liquidctl#322] [General]
@jonasmalacofilho I know you have mentioned before that you have no plans for a v2 of liquidctl, and I understand why. 

There are several deprecated commands / flags / options in liquidctl, do you think it would be worth saying what version those are planned to be removed in? Would they be removed in the potential v2.0, v2.1 or some other version? 

Im not necessarily saying that we should plan those versions but it might be worth giving a notice that they will be removed at some-point in the next major version bump if there is one. I think planning for it, but not actually doing it is worth it.

**Comment by jonasmalacofilho** [comment]
I'll document this, together with what exactly we try to guarantee between minor version bumps.

But, as a preview, deprecation is orthogonal to removal: deprecation is discouragement of use; on the other hand, anything that's not useful and that can be removed, will (generally) be removed as soon as possible.  In the end it's about what is a breaking change.

Also, we can't really schedule the removal of some features in a breaking vFUTURE, if we don't yet have defined what this vFUTURE would be.  It could be a _small_ breaking change, to fix a specific issue/limitation, or it could be a massive rewrite on the entire codebase (or anything in between); in the first case, we would still try to limit breaking changes to those related to the particular issue/limitation we would be addressing, but in the second we would remove or change everything, regardless of previous deprecation.

---

P.S. This was probably better suited to an issue, since there's something actionable to do (~~plan~~ document).  At least given the limitation of this "Discussions" feature.

**Reply by MarshallAsch** [reply]
I wasn't sure, I was originally going to make it as an issue but then thought it might be worth discussing before making the todo task.

**Reply by jonasmalacofilho** [reply]
Discussions on specific actionable things can still be made within an issue...

Or, at least, I think "Discussions" is for less on-topic or actionable conversation, like what could we do in an hypothetical v2.

**Reply by jonasmalacofilho** [reply]
Since this is not an issue, please ping me in two weeks if it seems that I forgot about it.

_(I want to add breaking changes and deprecation as sections to a new CONTRIBUTING document)._

**Reply by jonasmalacofilho** [reply]
Done, see [`docs/developer/process.md`](https://github.com/liquidctl/liquidctl/blob/main/docs/developer/process.md) (and [`CONTRIBUTING.md`](https://github.com/liquidctl/liquidctl/blob/main/CONTRIBUTING.md)).
